storage/cloud-client/storage_set_public_access_prevention_enforced.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+# [START storage_set_public_access_prevention_enforced]
+from google.cloud import storage
+from google.cloud.storage.bucket import PUBLIC_ACCESS_PREVENTION_ENFORCED
+
+
+def set_public_access_prevention_enforced(bucket_name):
+    """Enforce public access prevention for a bucket."""
+    # bucket_name = "my-bucket"
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+
+    bucket.iam_configuration.public_access_prevention = PUBLIC_ACCESS_PREVENTION_ENFORCED
```

Comment:
I would recommend switching to string literals instead of constants because they're not used actively in our samples. 

I may have missed samples that are now using string constants though. I would still stay away from it unless it's a strict requirement.

Reply:
Other samples in this library (such as `change_default_storage_class`) are using string constants. Why do you suggest staying away from it? I believe this is a safer implementation. Someone using this constant from the library will not have to change their code if we change the value of this constant down the line.

Reply:
I'm making a change to address the usage of constants in that sample here: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/5329

The constants.py isn't intentional about how a developer is going to use them. I agree that this is something we should do (rely on constants provided by the library), but we need to audit the existing constants experience.

For example, some are located [constants.py](https://github.com/googleapis/python-storage/blob/master/google/cloud/storage/constants.py) and others are located here: [iam.py](https://github.com/googleapis/python-storage/blob/master/google/cloud/storage/iam.py#L22-L41). For PAP, they were added [here](https://github.com/googleapis/python-storage/blob/f9498b982d5011971ca7809ab279cf738bcc07f1/google/cloud/storage/bucket.py#L89).

I agree this makes sense for supportability reasons as you stated, but we should be intentional / guided about where these constants exist which in turn will make it easier for developers to find them. Otherwise, it's a maze.

I might be too cautious as well, but samples have used string literals for the most part.